pkg/controller/secretsmanager/secret/setup.go

```diff
@@ -172,18 +258,13 @@ func (e *hooks) isUpToDate(cr *svcapitypes.Secret, resp *svcsdk.DescribeSecretOu
 	case len(s.SecretBinary) != 0:
 		return bytes.Equal(payload, s.SecretBinary), nil
 	}
-	return false, errors.New("neither SecretString nor SecretBinary field is filled in the returned object")
+	return false, errors.New(errNoAWSValue)
 }
 
-func (e *hooks) getPayload(ctx context.Context, cr *svcapitypes.Secret) ([]byte, error) {
-	var ref *svcapitypes.SecretReference
-	switch {
-	case cr.Spec.ForProvider.StringSecretRef != nil:
-		ref = cr.Spec.ForProvider.StringSecretRef
-	case cr.Spec.ForProvider.BinarySecretRef != nil:
-		ref = cr.Spec.ForProvider.BinarySecretRef
-	default:
-		return nil, errors.New("neither binarySecretRef nor stringSecretRef is given")
+func (e *hooks) getPayload(ctx context.Context, cr *svcapitypes.SecretParameters) ([]byte, error) {
+	ref, err := getSecretRef(cr)
```

Comment:
I think it's easier to follow what's happening if we have duplication for that switch case instead of calling another function.

Reply:
I would prefer to keep code encapsulated that is used more than once.

I modified the function so it will return an error in case `StringSecretRef` and `BinarySecretRef` are both set. Before that, `StringSecretRef` took preference over `BinarySecretRef` and I don't think that was intended. I also added a note to the API docs.